custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -103,14 +105,55 @@ def __init__(self, model_dir, target_type=None):
         # New custom task class and instance loaded from custom.py
         self._custom_task_class = None
         self._custom_task_class_instance = None
+        self._guard_moderation_hooks = None
+        self._guard_pipeline = None
 
         if target_type == TargetType.TEXT_GENERATION:
             self._target_name = os.environ.get("TARGET_NAME")
             if not self._target_name:
                 raise ValueError("Unexpected empty target name for text generation!")
+            self._load_guard_hooks_for_drum()
         else:
             self._target_name = None
 
+    def _load_guard_hooks_for_drum(self):
+        if os.environ.get("MLOPS_GUARD_HOOK_FILE", None) is None:
+            return
+
+        guard_file_prefix = os.environ.get("MLOPS_GUARD_HOOK_FILE")
+        guards_file_paths = list(Path(self._model_dir).rglob("{}.py".format(guard_file_prefix)))
+        if len(guards_file_paths) > 1:
+            self._logger.error("Found too many guard hook files: {}".format(guards_file_paths))
+            return
+
+        if len(guards_file_paths) == 0:
+            self._logger.info(
+                "No {}.py file detected in {}".format(guard_file_prefix, self._model_dir)
+            )
+            return
+
+        guard_file_path = guards_file_paths[0]
+        self._logger.info("Detected {} .. trying to load hooks".format(guard_file_path))
+        sys.path.insert(0, os.path.dirname(guard_file_path))
+
+        try:
+            guard_module = __import__(guard_file_prefix)
+            self._guard_moderation_hooks = {
+                GUARD_INIT_HOOK_NAME: getattr(guard_module, GUARD_INIT_HOOK_NAME, None),
+                GUARD_SCORE_WRAPPER_NAME: getattr(guard_module, GUARD_SCORE_WRAPPER_NAME, None),
+            }
+            if (
+                self._guard_moderation_hooks[GUARD_INIT_HOOK_NAME]
+                and self._guard_moderation_hooks[GUARD_SCORE_WRAPPER_NAME]
+            ):
+                self._guard_pipeline = self._guard_moderation_hooks[GUARD_INIT_HOOK_NAME]()
```

Comment:
Why var is called `_guard_pipeline` if you assign an `init` hook?

Reply:
If you notice carefully, it is not `init` hook being assigned, but `init` hook being "called" "()" at the end.  The init hook will return the the `pipeline` object.